Library/Homebrew/dev-cmd/contributions.rb

```diff
@@ -19,11 +19,11 @@ module Homebrew
   sig { returns(CLI::Parser) }
   def contributions_args
     Homebrew::CLI::Parser.new do
-      usage_banner "`contributions` <email|name> [<--repositories>`=`] [<--csv>]"
+      usage_banner "`contributions` <email|username> [<--repositories>`=`] [<--csv>]"
```

Comment:
Maybe this should only ever take the username for consistency?

Reply:
Cool. I guess the problem I'm hoping to figure out a solution to (even if a manual one) is: how do I combine the results from the various different outputs into a cohesive one without double counting?

Reply:
> or some querying of our own to filter to get them

Can we ask the API for email addresses associated with a user name and then use those to filter `git log`?

Reply:
> > or some querying of our own to filter to get them
> 
> 
> 
> Can we ask the API for email addresses associated with a user name and then use those to filter `git log`?

Nope, only the authenticated user can get their own "private" email addresses: https://docs.github.com/en/rest/users/emails?apiVersion=2022-11-28#list-email-addresses-for-the-authenticated-user. Which makes sense.

I suppose what we _could_ do is get all the commits for an author, slurp the email addresses and then, discounting the commits we already have, scan again for coauthors and signoffs ourselves? But that feels like a lot.

Reply:
Can we query the API for public email addresses? Maybe that will be enough to work with.

Reply:
Yes, but there's only ever one "primary" email address. 🤔

Reply:
> Cool. I guess the problem I'm hoping to figure out a solution to (even if a manual one) is: how do I combine the results from the various different outputs into a cohesive one without double counting?

To perhaps be even more explicit (and this addresses backwards compatibility too): my understanding is the goal of this command is to be able to have me, and ideally any maintainer, see which maintainers are likely to be asked to step down due to inactivity.

Personally, I think that means:
- there's no need for any backwards compatibility
- a single path and fewer options are better, even if the output is inferior and if we need e.g. to run against both email and username and check both: this tool should be doing the aggregation and not the user

Given this: I think it'd make sense for something like a bare `brew contributions` (i.e. the default options) to spit out the contributions for:
- all current maintainers
- over the last year
- using their username and/or email
- on primary repositories, grouped by repo
- sorted from most to least active

@issyl0 I can privately share the spreadsheet I filled in manually to calculate this last time if it helps.

Reply:
Thanks, that'd be useful. This definitely needs work. Sorry for the back and forth/confusion/diversions. The last "makes sense for" gives me more to work with, thanks!

Reply:
@issyl0 Absolutely no problem on back-and-forth here! I've not been terribly clear/explicit here and wasn't sure if we were agreed on the goals here!

Reply:
Gonna post that comment in the main tracking isue and merge this as-is, since it's useful (and more accurate) than the previous implementation (_for a GitHub username_). Then I'll iterate on the `maintainers` stuff more soon.